### DISCUSSION TITLE
Autogenerated examples force many changes due to generated data

### DISCUSSION THREAD

**Discussion by dgmora** [rswag/rswag#531] [General]
Hi, I have a doubt regarding autogenerated examples.

Because we use [Faker](https://github.com/faker-ruby/faker) to generate random data, when we re-run multiple times rspec, we generate new examples (since the data is different in every run). I managed to reduce the impact of this by using the same seed within Faker, but I it's not really possible to avoid that for timestamps like `created_at`.

So whenever I re-rerun rspec, I have changes in the definition:
<img width="367" alt="Screenshot 2022-07-12 at 15 34 37" src="https://user-images.githubusercontent.com/3609616/178502623-2c554086-7b1f-463a-ba46-bc22c79bba8b.png">

Is there a way to avoid this? Is it possible to ignore certain fields to be updated? Would a PR be accepted for something like that?

**Comment by dollerbill** [comment]
@dgmora I had the same issue, my solution so far has been taking my Faker data and creating fixtures for certain objects so that they would stop generating new time values, IDs, etc.

**Comment by dgmora** [comment]
The problem isn't Faker. Since we create new objects for each test,
timestamps are new every time

**Comment by BookOfGreg** [comment]
Could always use the built-in time stub travel_to from the ActiveSupport::Testing::TimeHelpers section.
https://api.rubyonrails.org/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-travel_to

**Comment by jtannas** [comment]
Seems resolved. Closing.
